### PR TITLE
UsernameUsed Test case and temp user helper method

### PR DESCRIPTION
**Pull request by DLzer**
Still learning here, but let me know if this is correct formatting. I did two things:

1. Created a helper function for inserting a temp user. Assuming multiple tests will require this.
2. Created an initial test case for checking against a username that's already in use.

### CODE REVIEW

**Comment by incidrthreat** [comment]
I like the initiative of creating a helper func to input a user in the db, nice touch.  The UsernameUsed test case needs some work though.  In the login test case we didn't care to catch the token being returned, but in the UsernameUsed we are concerned with the server response and what that means in addition to the errors.  We'll want to expand on that test case for a username in use and one that is not.

Note: The response from the grpc service when calling UsernameUsed is a boolean response.  So use that to build that portion of the tests. If true on non-existent user, the test fails.  If true on existing user, test passes.

Not sure if that makes sense, but reach out if you need a better, non-confusing, explanation.

**Comment by DLzer** [comment]
@incidrthreat Made some adjustments. Not sure if it's any better. Unsure of the response fomrat for UsernameUsed.

**Comment by DLzer** [comment]
@incidrthreat @kkweon Good to go?

File: authservice/cmd/authservice/authservice_test.go [hunk 1]
```diff
@@ -33,8 +41,36 @@ func Test_authServer_Login(t *testing.T) {
...
+	}
+
+	res, err = server.UsernameUsed(context.Background(), &pb.UsernameUsedRequest{Username: "testUser"})
+	// 3. Server responded with an error
+	if err != nil {
+		t.Error("3. An error was returned: ", err.Error())
+	}
+	// 3. Our username does not exist, If the server responded [false] which it shouldn't.. the test fails
+	if res.GetUsed() {
+		t.Error("3. User name is not used, should have returned false")
```
Comment (kkweon) [inline authservice/cmd/authservice/authservice_test.go:68]:
```suggestion
		t.Error("4. User name is not used, should have returned false")
```

File: authservice/cmd/authservice/authservice_test.go [hunk 2]
```diff
@@ -33,8 +41,36 @@ func Test_authServer_Login(t *testing.T) {
...
+	if !res.GetUsed() {
+		t.Error("2. Username is used, should have returned true")
+	}
+
+	res, err = server.UsernameUsed(context.Background(), &pb.UsernameUsedRequest{Username: "testUser"})
+	// 3. Server responded with an error
+	if err != nil {
+		t.Error("3. An error was returned: ", err.Error())
+	}
+	// 3. Our username does not exist, If the server responded [false] which it shouldn't.. the test fails
```
Comment (kkweon) [inline authservice/cmd/authservice/authservice_test.go:66]:
```suggestion
	// 4. Our username does not exist, If the server responded [false] which it shouldn't.. the test fails
```

**Pull request merged**